### ISSUE TITLE
How to deploy configuration to use my own ssl cert file?

### ISSUE THREAD

**Issue by phanmemkhoinghiep** [antoniomika/sish#159]
I already have my own cert file
How to deploy configuration to use it?
Tks

**Comment by antoniomika** [comment]
Hi there, could you clarify what type of cert? SSL cert, SSH cert, etc?

**Comment by phanmemkhoinghiep** [comment]
> 
> 
> Hi there, could you clarify what type of cert? SSL cert, SSH cert, etc?

Its SSH cert for connection from HTTPS client to SISH server (The HTTPS Server in local side)
Another question, SISH provice both of HTTP & HTTPS connection, how can I turn off HTTP connection and using HTTPS only with my own SSL cert
Tks!

**Comment by phanmemkhoinghiep** [comment]
2021/04/17 - 07:33:30 | HTTP forwarding started: http://subdomain.domain -> /tmp/XXX.XXX.XX.XXX:54556:80497672657 for client: 123.17.147.185:54556
2021/04/17 - 07:33:30 | HTTPS forwarding started: https://subdomain.domain -> /tmp/XXX.XXX.XX.XXX:54556:80497672657 for client: 123.17.147.185:54556
http: TLS handshake error from XXX.XXX.XX.XXX:51552: ondemand certificate retrieval is not enabled

**Comment by phanmemkhoinghiep** [comment]
Finally, i solved by myself. Tks in advance

**Comment by bulletprof** [comment]
@phanmemkhoinghiep Can you please share how did you solve it?

**Comment by bsnuggs1** [comment]
> [@phanmemkhoinghiep](https://github.com/phanmemkhoinghiep) Can you please share how did you solve it?

4 years late, but I ran into the same issue.  Hopefully, this helps someone.

----

I was running into a similar problem with getting it to work.  I was able to fix my issue by making sure that the certificate and key uses the exact notation requested in the CLI help.  In the [CLI help](https://github.com/antoniomika/sish/blob/87f8d8e819ec5928012880eb9cd7cbe74abd868a/docs/posts/cli.md?plain=1#L76), it says

```
The directory containing HTTPS certificate files (name.crt and name.key). There can be many crt/key pairs (default "deploy/ssl/")
```
Relevant code: https://github.com/antoniomika/sish/blob/main/utils/utils.go#L231-L245  

----

My certificates from dnsrobocert were cert.pem and privkey.pem, so whenever I had bind mounted the certs to the SSL folder, it would never read them.  I added the below deploy_hook to my letsencrypt config, added a new bind mount for the folder, and then it worked when I re-upp'd the compose:

```
deploy_hook: >-
    /bin/cp /etc/letsencrypt/live/example.com/privkey.pem /ssl/example.com.key;
    /bin/cp /etc/letsencrypt/live/example.com/cert.pem /ssl/example.com.crt;
```
